base_layer/core/src/base_node/service/service.rs

```diff
@@ -158,7 +158,7 @@ where B: BlockchainBackend + 'static
             executor,
             outbound_message_service,
             inbound_nch,
-            waiting_requests: HashMap::new(),
+            waiting_requests: WaitingRequests::new(),
```

Comment:
This might be fine as a local var in `start`

Reply:
I left this in the struct as it helped with cleaned the spawning code.